front50-sql/src/main/kotlin/com/netflix/spinnaker/front50/model/SqlStorageService.kt

```diff
@@ -164,20 +167,76 @@ class SqlStorageService(
     return objects
   }
 
+  override fun <T : Timestamped> loadObjectsNewerThan(
+    objectType: ObjectType,
+    lastModifiedThreshold: Long
+  ):
+    Map<String, List<T>> {
+
+    log.debug("Fetching {} objects with last_modified_at value greater than {}",
+      objectType.name,
+      lastModifiedThreshold)
+
+    val deletedKey = "deleted"
+    val notDeletedKey = "not_deleted"
+
+    val resultMap = mapOf<String, MutableList<T>>(
+      deletedKey to mutableListOf(),
+      notDeletedKey to mutableListOf()
+    )
+
+    val tableSupportsSoftDeletes = definitionsByType[objectType]!!.supportsHistory
+
+    val fieldsToFetch = mutableListOf<SelectFieldOrAsterisk>(field("body", String::class.java))
+    if (tableSupportsSoftDeletes) {
+      fieldsToFetch.add(field("is_deleted", Boolean::class.java))
+    }
+
+    val timeToLoadObjects = measureTimeMillis {
+      val result: Result<Record> = jooq.withRetry(sqlRetryProperties.reads) { ctx ->
+        ctx
+          .select(fieldsToFetch)
+          .from(definitionsByType[objectType]!!.tableName)
+          .where(field("last_modified_at", Long::class.java).greaterThan(lastModifiedThreshold))
```

Comment:
Think it's worth adding an index on this column?

Reply:
Done in https://github.com/spinnaker/front50/pull/1256.